src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs

```diff
@@ -20,81 +20,96 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
-                [ClassificationTypeNames.PreprocessorKeyword] = SemanticTokenTypes.Macro,
-                // in https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers
-                [ClassificationTypeNames.LabelName] = "label",
-            }).ToDictionary(kvp => kvp.Key, kvp => kvp.Value));
+        private static readonly ImmutableDictionary<string, string> s_pureLspDirectTypeMap = s_vsDirectTypeMap.Concat(new Dictionary<string, string>
+        {
+            [ClassificationTypeNames.ClassName] = SemanticTokenTypes.Class,
+            [ClassificationTypeNames.StructName] = SemanticTokenTypes.Struct,
+            [ClassificationTypeNames.NamespaceName] = SemanticTokenTypes.Namespace,
+            [ClassificationTypeNames.EnumName] = SemanticTokenTypes.Enum,
+            [ClassificationTypeNames.InterfaceName] = SemanticTokenTypes.Interface,
+            [ClassificationTypeNames.TypeParameterName] = SemanticTokenTypes.TypeParameter,
+            [ClassificationTypeNames.ParameterName] = SemanticTokenTypes.Parameter,
+            [ClassificationTypeNames.LocalName] = SemanticTokenTypes.Variable,
+            [ClassificationTypeNames.PropertyName] = SemanticTokenTypes.Property,
+            [ClassificationTypeNames.MethodName] = SemanticTokenTypes.Method,
+            [ClassificationTypeNames.EnumMemberName] = SemanticTokenTypes.EnumMember,
+            [ClassificationTypeNames.EventName] = SemanticTokenTypes.Event,
+            [ClassificationTypeNames.PreprocessorKeyword] = SemanticTokenTypes.Macro,
+            // in https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers
+            [ClassificationTypeNames.LabelName] = "label",
+        }).ToImmutableDictionary();
 
         /// <summary>
-        /// Mapping from roslyn <see cref="ClassificationTypeNames"/> to the LSP <see cref="SemanticTokenTypes"/> we
-        /// should use for them.  If something is not mapped, we will pass along the roslyn type name along.
+        /// A schema for mapping classification type names to VS LSP token names.  This maps a few classification type names
+        /// directly to LSP semantic token types, but otherwise generally returns the classification type name as a custom token type.
         /// </summary>
-        public readonly IReadOnlyDictionary<string, string> TokenTypeMap;
+        private static readonly SemanticTokensSchema s_vsTokenSchema = new(ClassificationTypeNames.AllTypeNames
+            .Where(classificationTypeName => !ClassificationTypeNames.AdditiveTypeNames.Contains(classificationTypeName))
+            .ToImmutableDictionary(
+                classificationTypeName => classificationTypeName,
+                classificationTypeName => IDictionaryExtensions.GetValueOrDefault(s_vsDirectTypeMap, classificationTypeName) ?? classificationTypeName));
```

Comment:
why are you calling the extension in full form instead of extension form?

Reply:
There was an ambiguous reference error between this and a readonly version